### PR TITLE
Fixes granted by cascade/restrict statements for revoke

### PR DESCRIPTION
**Pull request by gurkanindibay**
DESCRIPTION: Fixes incorrect propagating of `GRANTED BY` and `CASCADE/RESTRICT` clauses for `REVOKE` statements

There are two issues fixed in this PR
1. granted by statement will appear for revoke statements as well
2. revoke/cascade statement will appear after granted by

Since granted by statements does not appear in statements, this bug hasn't been visible until now. However, after activating the granted by statement for revoke, order problem arised and this issue was fixed order problem for cascade/revoke as well
In summary, this PR provides usage of granted by statements properly now with the correct order of statements.
We can verify the both errors, fixed with just single statement
REVOKE dist_role_3 from non_dist_role_3 granted by test_admin_role cascade;

### CODE REVIEW

File: src/test/regress/sql/create_role_propagation.sql [hunk 1]
```diff
@@ -137,7 +137,7 @@ REVOKE dist_role_3 from dist_role_4 granted by test_admin_role;
 SELECT roleid::regrole::text AS role, member::regrole::text, (grantor::regrole::text IN ('postgres', 'non_dist_role_1', 'dist_role_1','test_admin_role')) AS grantor, admin_option FROM pg_auth_members WHERE roleid::regrole::text LIKE '%dist\_%' ORDER BY 1, 2;
 SELECT objid::regrole FROM pg_catalog.pg_dist_object WHERE classid='pg_authid'::regclass::oid AND objid::regrole::text LIKE '%dist\_%' ORDER BY 1;
 
-REVOKE dist_role_3 from non_dist_role_3 granted by test_admin_role;
```
Comment (onurctirtir) [inline src/test/regress/sql/create_role_propagation.sql:140]: Rather than changing an existing test case that already works nicely before the fix, it'd be great to add a test that would fail on main but succeeds on this branch etc.

Comment (gurkanindibay) [reply src/test/regress/sql/create_role_propagation.sql:140]: This error is tricky actually. Since granted by statement is not being propagated, revoke statement never parsed before; therefore the error never exists. This appeared after I need to use granted by statement in revoke statement with cascade. Since there is no such test in main branch, there has been no errors actually.

File: src/test/regress/expected/create_role_propagation.out [hunk 2]
```diff
@@ -259,7 +259,25 @@ SELECT result FROM run_command_on_all_nodes(
...
+    SELECT member::regrole::text, roleid::regrole::text AS role, grantor::regrole::text, admin_option
+    FROM pg_auth_members WHERE roleid::regrole::text = 'dist_role_3'
+  ) q;
+  $$
+);
+                                                result
+---------------------------------------------------------------------
+ [{"member":"non_dist_role_3","role":"dist_role_3","grantor":"test_admin_role","admin_option":false}, +
+  {"member":"test_admin_role","role":"dist_role_3","grantor":"postgres","admin_option":true}]
+ [{"member":"dist_role_4","role":"dist_role_3","grantor":"test_admin_role","admin_option":false},     +
```
Comment (onurctirtir) [inline src/test/regress/expected/create_role_propagation.out:275]: The following line disappeared from coordinator (see the first array in `result`) and but not from workers, do you think what's going wrong?

```sql
{"member":"dist_role_4","role":"dist_role_3","grantor":"test_admin_role","admin_option":false}
```

**Review by onurctirtir — COMMENTED** [review]
LGTM except the incorrectness in the test output.

File: src/test/regress/expected/create_role_propagation.out (hunk 2 above)
Comment (onurctirtir) [reply src/test/regress/expected/create_role_propagation.out:275]: Also, flaky test detection jobs are failing.

Comment (gurkanindibay) [reply src/test/regress/expected/create_role_propagation.out:275]: Yes I noticed right now. It is successful in local. Let me check it

Comment (onurctirtir) [reply src/test/regress/expected/create_role_propagation.out:275]: > The following line disappeared from coordinator (see the first array in `result`) and but not from workers, do you think what's going wrong?
> 
> ```sql
> {"member":"dist_role_4","role":"dist_role_3","grantor":"test_admin_role","admin_option":false}
> ```

Disappearing from coordinator's output (first array in the `result`) is the correct behavior but the fact that this doesn't apply to worker outputs doesn't sound correct.

Comment (gurkanindibay) [reply src/test/regress/expected/create_role_propagation.out]: Problem was I didn't perform a fresh build before getting test outputs. Now pushed test outputs with fresh builds

**Comment by gurkanindibay** [comment]
@onurctirtir kindly request your re-review

**Review by onurctirtir — APPROVED** [review]
